workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/src/Generation/Model/Details/DetailsStore.cs

```diff
@@ -117,35 +118,45 @@ private void PopulateBlittableMaps()
 
         private bool? CheckBlittable(IEnumerable<Field> fields)
```

Comment:
A field is either blittable, or not - it feels over-complicated to have 3 potential states for a result.
I want to reiterate that storing a mapping of `name->bool` also feels over-complicated when presence or absence in a `HashSet` represents the same information more succinctly.

Reply:
Will change this to a DFS instead of BFS

Reply:
After going down the rabbit hole a bit, and discussing this offline - a solution that involves dropping the tri-state logic would require building a DAG representation of the types, where nodes are types, and edges represent a field that is a user type (dependency between types). You then check nodes where (all dependencies already checked = true) and insert into a hashset if its blittable.

At the end of the day, this solution is probably more complex and is easier to get wrong (imo), than the naive solution currently implemented.